kasa/klapprotocol.py

```diff
@@ -0,0 +1,477 @@
+        for retry in range(retry_count + 1):
+            try:
+                return await self._execute_query(request, retry)
+            except aiohttp.ServerDisconnectedError as sdex:
+                if retry >= retry_count:
+                    _LOGGER.debug("Giving up on %s after %s retries", self.host, retry)
+                    raise SmartDeviceException(
+                        f"Unable to connect to the device: {self.host}: {sdex}"
+                    )
+                continue
+            except aiohttp.ClientConnectionError as ex:
+                raise SmartDeviceException(
+                    f"Unable to connect to the device: {self.host}: {ex}"
+                )
+            except TimeoutError as ex:
+                raise SmartDeviceException(
+                    f"Unable to connect to the device, timed out: {self.host}: {ex}"
+                )
+            except AuthenticationException as auex:
+                _LOGGER.debug("Unable to authenticate with %s, not retrying", self.host)
+                raise auex
+            except Exception as ex:
+                if retry >= retry_count:
+                    _LOGGER.debug("Giving up on %s after %s retries", self.host, retry)
+                    raise SmartDeviceException(
+                        f"Unable to connect to the device: {self.host}: {ex}"
+                    )
+                continue
+
+        # make mypy happy, this should never be reached..
+        raise SmartDeviceException("Query reached somehow to unreachable")
+
+    async def _execute_query(self, request: str, retry_count: int) -> Dict:
+        timeout = aiohttp.ClientTimeout(total=self.timeout)
+
+        async with aiohttp.ClientSession(
+            cookie_jar=self.jar, timeout=timeout
+        ) as session:
+            if not self.handshake_done or self.handshake_session_expired():
+                try:
+                    await self.perform_handshake(session)
+
+                except AuthenticationException as auex:
+                    _LOGGER.debug(
+                        "Unable to complete handshake for device %s, authentication failed",
+                        self.host,
+                    )
+                    raise auex
+
+            # Check for mypy
+            if self.encryption_session is not None:
+                payload, seq = self.encryption_session.encrypt(request.encode())
+
+            url = f"http://{self.host}/app/request"
+
+            response_status, response_data = await self.session_post(
+                session, url, params={"seq": seq}, data=payload
+            )
+
+            msg = f"at {datetime.datetime.now()}.  Host is {self.host}, Retry count is {retry_count}, Sequence is {seq}, Response status is {response_status}, Request was {request}"
+            if response_status != 200:
+                if _LOGGER.isEnabledFor(logging.ERROR):
+                    _LOGGER.error("Query failed after succesful authentication " + msg)
```

Comment:
```suggestion
                _LOGGER.error("Query failed after succesful authentication %s", msg)
```
No need to guard any logger calls besides those whose argument generation produces overhead.

Reply:
Done